app/scripts/modules/core/src/application/application.state.provider.ts

```diff
@@ -78,8 +78,21 @@ export class ApplicationStateProvider implements IServiceProvider {
   public addParentState(parentState: INestedState, mainView: string, relativeUrl = '') {
     const applicationConfig: INestedState = {
       name: 'application',
-      abstract: true,
       url: `${relativeUrl}/:application`,
+      redirectTo: (transition) => {
+        return transition
+          .injector()
+          .getAsync('app')
+          .then((app: Application) => {
+            const environments = app.getDataSource('environments');
+            const defaultState = environments && !environments.disabled ? '.environments' : '.insight.clusters';
+
+            const params = transition.params();
+            const options = { relative: transition.to().name };
+
+            return transition.router.stateService.target(defaultState, params, options);
```

Comment:
I am in no way attached to this specific approach, it's modified [copy-pasta from the serverGroups state](https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/serverGroup/serverGroup.states.ts#L28-L43)

Reply:
This looks good to me, and accounts for for access to apps via projects and normally

Reply:
I like that is it centralized here.